app/controllers/pensions/UpdatePensionProviderController.scala

```diff
@@ -90,104 +89,111 @@ class UpdatePensionProviderController @Inject() (
   def doYouGetThisPension(): Action[AnyContent] = authenticate.authWithDataRetrieval.async { implicit request =>
```

Comment:
This doesn't need to be async. Remove futures and just call apply. Will be simpler. Same with some other methods, eg addTelephoneNumber